s3-media-sync.php

```diff
@@ -26,7 +26,13 @@
 	function() {
 		$settings_handler = new S3_Media_Sync_Settings();
 		$settings = $settings_handler->get_settings();
-		$tester = new S3_Media_Sync_Tester($settings);
+		
+		// Only create the tester if we have the required settings configured
+		// This prevents errors when the plugin is first activated with no settings
+		if ( $settings_handler->has_required_settings() ) {
+			$tester = new S3_Media_Sync_Tester($settings);
+		}
+		
 		$s3_media_sync = new S3_Media_Sync($settings_handler);
```

Comment:
Follow-up request: after activation with incomplete config, please add a dismissible `admin_notices` message linking to the settings screen so users know what to do next.